### PR TITLE
Workaround for #21

### PR DESCRIPTION
**Pull request by bbhopesh**
Workaround for longjmp [bug](https://github.com/LinusU/cwasm-jpeg-turbo/issues/2) in `cwasm-jpeg-turbo` The bug was originally reported in #21 

The workaround is to fallback on [jpeg-js](https://www.npmjs.com/package/jpeg-js)

### CODE REVIEW

**Comment by bbhopesh** [comment]
Hey @LinusU, can you please take a look a this?

**Comment by bbhopesh** [comment]
Hey @pwlmaciejewski  I have added the test case and updated the PR. The test suite is passing with latest changes, and the test suite fails without these changes. So, the change is good.

However, output of the test suite is weird when it fails. It shows some other test as failing even though the newly added test is the one that is actually failing.

File: test/index.spec.js [hunk 1]
```diff
@@ -59,6 +59,22 @@ it("should hash palette based pngs correctly", async () => {
...
 });
 
+/**
+ * For some jpeg images, cwasm-jpeg-turbo throws longjmp error due to a bug in WebAssembly.
+ * Added a workaround to use alternate libraries to read such images.
+ *
+ * see: https://github.com/LinusU/cwasm-jpeg-turbo/issues/2
+ * see: https://github.com/pwlmaciejewski/imghash/issues/21
+ */
+it("should not throw longjmp error", () => {
```
Comment (pwlmaciejewski) [inline test/index.spec.js:69]: Ok @bbhopesh, I see why it reports some other test failing.
Async function will be executed after the test execution. Or in other words, async/await doesn't go well with `.toThrow()`. Here's a general solution: https://github.com/facebook/jest/issues/1700#issuecomment-377890222

I would rewrite this test to something like this:

```js
it("should not throw longjmp error",  () => {
  expect(imghash.hash(__dirname + "/files/longjmperror.jpg", 8)).resolves.toHaveLength(16)
});
```

**Comment by pwlmaciejewski** [comment]
> However, output of the test suite is weird when it fails. It shows some other test as failing even though the newly added test is the one that is actually failing.

Thanks, @bbhopesh! I left you comment regarding this ☝️

**Comment by bbhopesh** [comment]
Hi @pwlmaciejewski I have changed as suggested by you but the issue doesn't go away. I will try to see if I need to change other tests too. Can you please also take a look?

**Review by pwlmaciejewski — APPROVED** [review]

**Pull request merged**